tasks/extract-binaries-from-image/extract-binaries-from-image.yaml

```diff
@@ -0,0 +1,74 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: extract-binaries-from-image
+  labels:
+    app.kubernetes.io/version: "0.1.0"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task that extracts binaries to be released on github.com from an image
+  params:
+    - name: image_binaries_path
+      type: string
+      description: Path inside the image where the binaries to extract are stored
+      default: "releases"
+    - name: binaries_dir
+      type: string
+      description: Path to store the binaries in the workspace
+      default: binaries
+    - name: snapshotPath
+      type: string
+      description: Path to the JSON string of the mapped Snapshot spec in the data workspace
+      default: snapshot.json
+    - name: subdirectory
+      description: Subdirectory inside the workspace to be used for storing the binaries
+      type: string
+      default: ""
+  workspaces:
+    - name: data
+      description: The workspace where the snapshot is stored. The extracted binaries will be stored here as well.
+  results:
+    - name: binaries_dir
+      type: string
+      description: The directory inside the workspace where the binaries are stored
+  steps:
+    - name: extract-binaries-from-image
+      image: quay.io/hacbs-release/release-utils:48a1cd6c939eda25382472135be4afe2c9b4b703
+      script: |
+        SNAPSHOT_SPEC_FILE="$(workspaces.data.path)/$(params.snapshotPath)"
+        if [ ! -f "${SNAPSHOT_SPEC_FILE}" ] ; then
+            echo "Error: No valid snapshot file was provided."
+            exit 1
+        fi
+        IMAGE_URL=$(jq -r '.components[0].containerImage' "${SNAPSHOT_SPEC_FILE}")
```

Comment:
I would suggest to explicitly add the empty string default with `// ""`. The reason is that `jq` will sometimes return `""` and sometimes `null` for missing fields. I wasn't able to figure out why. But consider this:
```
jq -r '.components[0].containerImage' <<< "{}"
```
It prints nothing (= empty string).

But now this:
```
echo $(jq -r '.components[0].containerImage' <<< "{}")
```
This will actually print `null`. And `null` is a non-empty string. So it's safer to have a default value of an actual empty string.

Reply:
Thank you for the suggestion! Added